lib/canonicalizer.js

```diff
@@ -1,23 +1,26 @@
 'use strict';
 
-var path = require('path').posix;
-var utils = require('./utils');
+const path = require('path').posix;
+const Utils = require('./utils');
 
-var Canonicalizer = function(hashAlgo) {
-  // node 0.11 urlencodes caret as %5E
-  var haveToFixCaretAtUrlParser = utils.parseUrl('^').href !== '^';
+// node 0.11 urlencodes caret as %5E
```

Comment:
Yes, but it is not strictly related to the ES6 refactor. And I think it's about that Node 0.11 _started_ to urlencode the caret, so we will have to keep that.

Reply:
I've checked it. With Node v0.10, haveToFixCaretAtUrlParser will be false. But with Node v0.11+, including the latest Node version, it will be true. So we can remove the haveToFixCaretAtUrlParser variable, but not the usage in join. Also I think there are no tests for this. :( I recommend fixing it separately.

Reply:
Fair enough, thanks for looking into it.